modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/swap/DiscountingKnownAmountPaymentPeriodPricer.java

```diff
@@ -120,4 +121,17 @@ public void explainPresentValue(KnownAmountPaymentPeriod period, RatesProvider p
     }
   }
 
+  //-------------------------------------------------------------------------
+  @Override
+  public MultiCurrencyAmount currencyExposure(KnownAmountPaymentPeriod period, RatesProvider provider) {
+    return MultiCurrencyAmount.of(CurrencyAmount.of(period.getCurrency(), presentValue(period, provider)));
+  }
+
+  @Override
+  public double currentCash(KnownAmountPaymentPeriod period, RatesProvider provider) {
```

Comment:
As the amount is known, we don't need the full provider, the valuation date is enough.